### PR TITLE
use new Source.queue to avoid memory leak

### PR DESCRIPTION
**Pull request by Roiocam**
## Motivation

the legacy usage of `Source.queue` has a memory leak issue, see: https://github.com/akka/akka/issues/25798, the newest API has been solved that.

This PR should cherry-pick to 1.0.x ?

### CODE REVIEW

**Review by He-Pin — APPROVED** [review]
Lgtm

**Review by laglangyue — APPROVED** [review]
LGTM

**Comment by He-Pin** [comment]
@Roiocam I was sending the same pr to https://github.com/SwissBorg/pekko-persistence-postgres/pull/32

I think we should check other pekko persistence plugins too.

File: core/src/main/scala/org/apache/pekko/persistence/jdbc/journal/dao/BaseDao.scala [hunk 1]
```diff
@@ -29,8 +29,8 @@ abstract class BaseDao[T] {
 
   def baseDaoConfig: BaseDaoConfig
 
-  val writeQueue: SourceQueueWithComplete[(Promise[Unit], Seq[T])] = Source
-    .queue[(Promise[Unit], Seq[T])](baseDaoConfig.bufferSize, OverflowStrategy.dropNew)
+  val writeQueue: BoundedSourceQueue[(Promise[Unit], Seq[T])] = Source
```
Comment (He-Pin) [inline core/src/main/scala/org/apache/pekko/persistence/jdbc/journal/dao/BaseDao.scala:32]: make it private now?

**Comment by Roiocam** [comment]
> @Roiocam I was sending the same pr to [SwissBorg/pekko-persistence-postgres#32](https://github.com/SwissBorg/pekko-persistence-postgres/pull/32)
> 
> I think we should check other pekko persistence plugins too.

I just checked and it seems that other plugins do not use queues for batching operations at writing events.

**Review by He-Pin — APPROVED** [review]
lgtm

**Comment by He-Pin** [comment]
@Roiocam yes, please squash merge and cherry pick to 1.0.x thanks.

**Comment by He-Pin** [comment]
Oops, it should be merged with squash...

**Comment by pjfanning** [comment]
https://github.com/search?q=org%3Aapache%20SourceQueueWithComplete&type=code -- there is other Pekko code that uses SourceQueueWithComplete. Should that be rewritten too? Even if most of it is in tests, those tests can be viewed as a form of documentation of the code.

**Comment by He-Pin** [comment]
If it's been used with `dropNew` strategy ,then it can be updated with the new implementation by @jrudolph

**Comment by Roiocam** [comment]
> https://github.com/search?q=org%3Aapache%20SourceQueueWithComplete&type=code -- there is other Pekko code that uses SourceQueueWithComplete. Should that be rewritten too? Even if most of it is in tests, those tests can be viewed as a form of documentation of how to use the code.

like He-pin says, only `dropNew` has this issue, but I will check it when I am free.

**Comment by He-Pin** [comment]
> > https://github.com/search?q=org%3Aapache%20SourceQueueWithComplete&type=code -- there is other Pekko code that uses SourceQueueWithComplete. Should that be rewritten too? Even if most of it is in tests, those tests can be viewed as a form of documentation of how to use the code.
> 
> like He-pin says, only `dropNew` has this issue, but I will check it when I am free.

I checked that already,  it was backpresure strategy in other cases, which the current new source.queue doesn't support.

**Pull request merged**